sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2851,24 +2878,54 @@ prefetch_impl(AccessorTy acc, OffsetT byte_offset, simd_mask<1> pred) {
                 "Implicit truncation of 64-bit byte_offset to 32-bit is "
                 "disabled. Use -fsycl-esimd-force-stateless-mem or explicitly "
                 "convert offsets to a 32-bit vector");
-  check_lsc_vector_size<NElts>();
   check_lsc_data_size<T, DS>();
   check_cache_hints<cache_action::prefetch, PropertyListT>();
+
+  constexpr size_t Alignment =
+      detail::getPropertyValue<PropertyListT, alignment_key>(sizeof(T));
```

Comment:
This line does not seem correct. For bytes/words the alignment should still be 4-bytes or more.
for block_load the caller of block_load_impl adds the default alignment 4-or-more to the property list: https://github.com/intel/llvm/blob/602fddfb8f9f64db00e0072f233a0eef1be4ae27/sycl/include/sycl/ext/intel/esimd/memory.hpp#L1353-L1355

I think both functions prefetch_impl and LSC-block_load_impl should expect that alignment is already added to the properties. Can you please do this for prefetch_impl and block_load_impl:
```suggestion
  constexpr size_t Alignment = get_property<alignment_key>().value;
```

Reply:
Sounds reasonable. Thanks for adding the extra assert to prefetch_impl().